sdk/typescript/src/cryptography/ed25519-keypair.ts

```diff
@@ -69,18 +79,12 @@ export class Ed25519Keypair implements Keypair {
     options?: { skipValidation?: boolean },
   ): Ed25519Keypair {
     const secretKeyLength = secretKey.length;
-    if (secretKeyLength !== 64) {
-      // Many users actually wanted to invoke fromSeed(seed: Uint8Array), especially when reading from keystore.
-      if (secretKeyLength === 32) {
-        throw new Error(
-          'Wrong secretKey size. Expected 64 bytes, got 32. Similar function exists: fromSeed(seed: Uint8Array)',
-        );
-      }
+    if (secretKeyLength !== 32) {
       throw new Error(
-        `Wrong secretKey size. Expected 64 bytes, got ${secretKeyLength}.`,
+        `Wrong secretKey size. Expected 32 bytes, got ${secretKeyLength}.`,
       );
     }
-    const keypair = nacl.sign.keyPair.fromSecretKey(secretKey);
+    const keypair = nacl.sign.keyPair.fromSeed(secretKey);
     if (!options || !options.skipValidation) {
```

Comment:
This is an attack vector if someone is able to call this method passing in the 64 bytes `privkey || pubkey` with skipValidaton = true.